src/renderer/shared/api/governance/off-chain/service/polkassemblyService.ts

```diff
@@ -1,43 +1,56 @@
-import { dictionary } from '@shared/lib/utils';
 import type { ChainId } from '@shared/core';
+import { dictionary } from '@shared/lib/utils';
 import type { IGovernanceApi } from '../lib/types';
 import { offChainUtils } from '../lib/off-chain-utils';
 
-// TODO: use callback to return the data, instead of waiting all at once
 export const polkassemblyService: IGovernanceApi = {
   getReferendumList,
   getReferendumDetails,
 };
 
+type PolkassemblyData = {
+  count: number;
+  posts: {
+    title: string;
+    post_id: string;
+  }[];
+};
 /**
  * Request referendum list without details 100 units of data each round
  * Ping their API to check "total" referendums and request remaining with Promise.allSettled
  * @param chainId chainId value
- * @return {Promise}
+ * @param callback returns portions of data
  */
-async function getReferendumList(chainId: ChainId): Promise<Record<string, string>> {
+async function getReferendumList(chainId: ChainId, callback: (data: Record<string, string>) => void) {
   const chainName = offChainUtils.getChainName(chainId);
-  if (!chainName) return {};
 
-  const getApiUrl = (page: number, size = 100): string => {
-    return `https://api.polkassembly.io/api/v1/listing/on-chain-posts?proposalType=referendums_v2&page=${page}&listingLimit=${size}`;
-  };
+  if (chainName) {
+    const getApiUrl = (page: number, size = 100): string => {
+      return `https://api.polkassembly.io/api/v1/listing/on-chain-posts?proposalType=referendums_v2&page=${page}&listingLimit=${size}`;
+    };
 
-  try {
     const headers = new Headers();
     headers.append('x-network', chainName);
+    headers.append('Cache-Control', 'public, max-age=600, must-revalidate');
 
-    const ping = await (await fetch(getApiUrl(1, 1), { method: 'GET', headers })).json();
-    const iterations = Math.ceil(ping.count / 100);
-    const remainders = await offChainUtils.getRemainingReferendums(iterations, getApiUrl, { headers });
-    const referendums = [...ping.posts, ...remainders.flatMap((r) => r.posts)];
+    fetch(getApiUrl(1), { method: 'GET', headers })
+      .then((res) => res.json())
+      .then((ping: PolkassemblyData) => {
+        callback(parsePolkassemblyData(ping));
 
-    return dictionary(referendums, 'post_id', (item) => item.title);
-  } catch {
-    return {};
+        for (let index = 2; index <= Math.ceil(ping.count / 100); index++) {
```

Comment:
Maybe move `Math.ceil(ping.count / 100)` to function to better understand what happened here?